### ISSUE TITLE
Should `Style/YodaCondition` treat constants like literals?

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#11511]
Currently, `Style/YodaCondition` flags something like this:

```ruby
if 'red' == color
  foo
elsif 'green' == color
  bar
end
```

but not something like this:
```ruby
if Colors::RED == color
  foo
elsif Colors::GREEN == color
  bar
end
```

Should this cop treat constants like literals? (Or is there some inherent danger in doing so?)

**Comment by koic** [comment]
Yeah, Yoda conditions targets constant but only handles literal now unexpected. I've opened #11512 that makes constant aware.
